agents/rust/mediator/src/aries_agent/mod.rs

```diff
@@ -166,15 +166,16 @@ impl<T: BaseWallet + 'static, P: MediatorPersistence> Agent<T, P> {
     pub async fn auth_and_get_details(
         &self,
         sender_verkey: &Option<VerKey>,
-    ) -> Result<(String, VerKey, AriesDidDoc), String> {
+    ) -> Result<(String, VerKey, VerKey, AriesDidDoc), String> {
```

Comment:
These tuple are starting grow over our heads, it's not easy to see what the function returns without studying the actual implementation. Consider creating "response" structs for these internal responses.